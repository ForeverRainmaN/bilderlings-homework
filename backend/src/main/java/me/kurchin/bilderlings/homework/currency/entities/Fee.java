package me.kurchin.bilderlings.homework.currency.entities;

public class Fee {

  private long id;
  private String from;
  private String to;
  private double fee;

  public Fee(long id, String from, String to, double fee) {
    this.id = id;
    this.from = from;
    this.to = to;
    this.fee = fee;
  }

  public long getId() {
    return id;
  }

  public String getFrom() {
    return from;
  }

  public String getTo() {
    return to;
  }

  public double getFee() {
    return fee;
  }

  public boolean isSameCurrencyPair(Fee otherFee) {
    return this.from.equals(otherFee.getFrom())
        && this.to.equals(otherFee.getTo());
  }

  @Override
  public String toString() {
    return "Fee{" +
        "id=" + id +
        ", from='" + from + '\'' +
        ", to='" + to + '\'' +
        ", fee=" + fee +
        '}';
  }
}
