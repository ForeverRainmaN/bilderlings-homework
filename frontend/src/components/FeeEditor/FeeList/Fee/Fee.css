.bilderlings-homework-fee {
  display: flex;
  justify-content: space-between;
  margin-bottom: 10px;
}

.bilderlings-homework-fee-element {
  display: flex;
  justify-content: center;
  flex-direction: column;
  text-align: center;
}
