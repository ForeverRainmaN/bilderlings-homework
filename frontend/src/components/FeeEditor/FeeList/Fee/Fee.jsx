import React from 'react';
import PropTypes from 'prop-types'
import {Button} from "@blueprintjs/core";
import './Fee.css';

export default function Fee({from, to, fee, removeFee}) {
  return (
      <div className="bilderlings-homework-fee">
        <div className="bilderlings-homework-fee-element">{from}</div>
        <div className="bilderlings-homework-fee-element">{to}</div>
        <div className="bilderlings-homework-fee-element">{fee}</div>
        <Button className="bp3-intent-danger"
                onClick={removeFee}
        >
          Remove
        </Button>
      </div>
  );
}

Fee.propTypes = {
  from: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
  fee: PropTypes.number.isRequired,
  removeFee: PropTypes.func.isRequired
};
