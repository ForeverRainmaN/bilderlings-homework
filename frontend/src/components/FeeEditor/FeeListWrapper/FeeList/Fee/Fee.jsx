import React from 'react';
import PropTypes from 'prop-types'
import { Button } from "@blueprintjs/core";
import './Fee.css';

export default function CurrencyPair({from, to, fee, removeFee}) {
  return (
      <div className="bilderlings-homework-curr-pair">
        <span>{from}</span>
        <span>{to}</span>
        <span>{fee}</span>
        <Button className="bp3-intent-danger"
                onClick={removeFee}
        >
          Remove
        </Button>
      </div>
  );
}

CurrencyPair.propTypes = {
  from: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
  fee: PropTypes.number.isRequired
};