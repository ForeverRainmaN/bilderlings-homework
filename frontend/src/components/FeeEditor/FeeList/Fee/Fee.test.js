import React from 'react';
import ShallowRenderer from 'react-test-renderer';
import Fee from './Fee';

it('tests if fee renders correctly', () => {
  const tree = ShallowRenderer
  .create(<Fee
      from="USD"
      to="EUR"
      fee={0.05}
      removeFee={() => {}}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});