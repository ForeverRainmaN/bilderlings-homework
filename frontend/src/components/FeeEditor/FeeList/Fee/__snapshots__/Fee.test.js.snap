// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="bilderlings-homework-curr-pair"
>
  <span>
    USD
  </span>
  <span>
    EUR
  </span>
  <span>
    0.05
  </span>
  <button
    className="bp3-button bp3-intent-danger"
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    type="button"
  >
    <span
      className="bp3-button-text"
    >
      Remove
    </span>
  </button>
</div>
`;
